docs/cookbook/localTimeForFutureEvents.mjs

```diff
@@ -28,19 +28,21 @@ const tc39meetings = [
 
 // To follow the meetings remotely from Tokyo, calculate the times you would
 // need to join:
-const localTimeZone = Temporal.TimeZone.from('Asia/Tokyo');
 const localTimes = tc39meetings.map(({ dateTime, timeZone }) => {
-  return Temporal.DateTime.from(dateTime).inTimeZone(timeZone, { disambiguation: 'reject' }).inTimeZone(localTimeZone);
+  return Temporal.LocalDateTime.from({ ...Temporal.DateTime.from(dateTime).getFields(), timeZone }).with(
```

Comment:
Oops, I reversed the intent of this sample. I'm not convinced about returning `DateTime` though.  `LocalDateTime` has almost the the same API as `DateTime`, but it's safer to use. With data that's explicitly associated with a time zone (like this data) I think leaving it as `LocalDateTime` is a good idea. For example, if the user needs to do something time-zone-sensitive with the results (e.g. "Alexa, wake me up one hour before each TC39 meeting")  then it's also better to have it in `LocalDateTime` to avoid bugs. 

So unless you disagree, I'll rewrite this sample to return the results in the user's time zone but leave as `LocalTimeZone`.

Reply:
> Also the usage of `...getFields()` suggests to me that there ought to be an API to combine DateTime+TimeZone into LocalDateTime just like there is for Absolute+TimeZone.

There's `DateTime.prototype.toLocalDateTime(tz, options)` in the proposal. Do you think that's enough?  If not, do you think that`with` and `from` should accept a `dateTime` "field" ?  Or do you think a dedicated method e.g. `withDateTime` is needed?

Reply:
Oh, I missed that one. In that case I'd write this as `dateTime.toLocalDateTime(timeZone)` instead of using from() and the spread operator.

Reply:
> I'd write this as `dateTime.toLocalDateTime(timeZone)` instead of using from() and the spread operator.

Done. 

> there ought to be an API to combine DateTime+TimeZone into LocalDateTime just like there is for Absolute+TimeZone.

Let me know what you think about #720 for this purpose.